src/uuid/UUIDv6.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "UUIDv6.h"
+#include "spdlog/spdlog.h"
+
+namespace uprotocol::uuid {
+/**
+ * @brief Constructor for building UUIDv6
+*/
+UUIDv6::UUIDv6() {
+    generateUUID(uuidV6_);
+}
+/**
+ * @brief This method has the logic to compute and generates UUIDv6
+ * @param[out] uuid - contains UUIDv6 so that calling function will have the unique id.
+ * @note Here constructor calls this method to set member @uuidV6_
+*/
+void UUIDv6::generateUUID(uuid_t uUIDV6) {
+    uint64_t u64UT;
+    uint8_t* pu8UP;
+
+    uuid_generate_time_safe(uUIDV6);
+    if (-1 == uuid_generate_time_safe(uUIDV6)) {
```

Comment:
why do you do it twice?

Reply:
Good catch. I just retained the old logic. Tested with one api call works.